mailpoet/assets/js/src/common/premium_banner_with_upgrade/premium_banner_with_upgrade.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import MailPoet from 'mailpoet';
+import PremiumRequired from 'common/premium_required/premium_required';
+import Button from 'common/button/button';
+import ReactStringReplace from 'react-string-replace';
+
+type Props = {
+  message: React.ReactNode;
+  actionButton: React.ReactNode;
+};
+
+interface BannerWindow extends Window {
+  mailpoet_has_valid_api_key: boolean;
+  mailpoet_has_valid_premium_key: boolean;
+  mailpoet_subscribers_count: number;
+  mailpoet_subscribers_limit: number | boolean;
+  mailpoet_subscribers_limit_reached: boolean;
+  mailpoet_premium_active: boolean;
+  mailpoet_premium_plugin_installed: boolean;
+  mailpoet_premium_plugin_download_url: string;
+  mailpoet_premium_plugin_activation_url: string;
+  mailpoet_plugin_half_key: string;
+}
+
+declare let window: BannerWindow;
```

Comment:
We have this task: https://mailpoet.atlassian.net/browse/MAILPOET-2943

I am not sure if it is outdated or still valid and the team still wants to go in that direction. If we don't want to add the window usage to the new code so that we don't have to remove it later. Would it make sense to move these properties to the `MailPoet` object and use it from there?

Reply:
@pavel-mailpoet Noted 👍